Support tasty-1.5

```
tasty-silver-3.3.1.1$ cabal build -c 'tasty>=1.5'
[4 of 6] Compiling Test.Tasty.Silver.Interactive

Test/Tasty/Silver/Interactive.hs:581:25: error: [GHC-83865]
    • Couldn't match type: Ap (Reader Level) TestOutput
                     with: [Ap (Reader Level) TestOutput]
      Expected: OptionSet
                -> TestName
                -> [Ap (Reader Level) TestOutput]
                -> Ap (Reader Level) TestOutput
        Actual: OptionSet
                -> TestName
                -> Ap (Reader Level) TestOutput
                -> Ap (Reader Level) TestOutput
    • In the expression: handleGroup
      In a record update at fields ‘foldSingle’, ‘foldGroup’,
      with type constructor ‘TreeFold’
      and data constructor ‘TreeFold’.
      In the first argument of ‘foldTestTree’, namely
        ‘trivialFold
           {foldSingle = handleSingleTest, foldGroup = handleGroup}’
    |
581 |           , foldGroup = handleGroup
    |                         ^^^^^^^^^^^

Test/Tasty/Silver/Interactive.hs:928:31: error: [GHC-83865]
    • Couldn't match expected type: Int -> Maximum Int
                  with actual type: [Int -> Maximum Int]
    • In the first argument of ‘(.)’, namely ‘m’
      In the expression: m . (+ indentSize)
      In the expression: \ _ _ m -> m . (+ indentSize)
    |
928 |       , foldGroup = \_ _ m -> m . (+ indentSize)
    |                               ^
Error: cabal: Failed to build tasty-silver-3.3.1.1.
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -6,11 +6,11 @@
 #
 #   haskell-ci regenerate
 #
-# For more information, see https://github.com/haskell-CI/haskell-ci
+# For more information, see https://github.com/andreasabel/haskell-ci
 #
-# version: 0.15.20220609
+# version: 0.17.20230827
 #
-# REGENDATA ("0.15.20220609",["github","tasty-silver.cabal"])
+# REGENDATA ("0.17.20230827",["github","tasty-silver.cabal"])
 #
 name: Haskell-CI
 on:
@@ -36,14 +36,24 @@ jobs:
     strategy:
       matrix:
         include:
-          - compiler: ghc-9.4.0.20220523
+          - compiler: ghc-9.8.0.20230822
             compilerKind: ghc
-            compilerVersion: 9.4.0.20220523
+            compilerVersion: 9.8.0.20230822
             setup-method: ghcup
             allow-failure: true
-          - compiler: ghc-9.2.3
+          - compiler: ghc-9.6.2
             compilerKind: ghc
-            compilerVersion: 9.2.3
+            compilerVersion: 9.6.2
+            setup-method: ghcup
+            allow-failure: false
+          - compiler: ghc-9.4.7
+            compilerKind: ghc
+            compilerVersion: 9.4.7
+            setup-method: ghcup
+            allow-failure: false
+          - compiler: ghc-9.2.8
+            compilerKind: ghc
+            compilerVersion: 9.2.8
             setup-method: ghcup
             allow-failure: false
           - compiler: ghc-9.0.2
@@ -109,19 +119,20 @@ jobs:
           apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
           if [ "${{ matrix.setup-method }}" = ghcup ]; then
             mkdir -p "$HOME/.ghcup/bin"
-            curl -sL https://downloads.haskell.org/ghcup/0.1.17.8/x86_64-linux-ghcup-0.1.17.8 > "$HOME/.ghcup/bin/ghcup"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.19.5/x86_64-linux-ghcup-0.1.19.5 > "$HOME/.ghcup/bin/ghcup"
             chmod a+x "$HOME/.ghcup/bin/ghcup"
-            if $HEADHACKAGE; then "$HOME/.ghcup/bin/ghcup" config add-release-channel https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.7.yaml; fi
+            "$HOME/.ghcup/bin/ghcup" config add-release-channel https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.7.yaml;
             "$HOME/.ghcup/bin/ghcup" install ghc "$HCVER" || (cat "$HOME"/.ghcup/logs/*.* && false)
-            "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
+            "$HOME/.ghcup/bin/ghcup" install cabal 3.10.1.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
           else
             apt-add-repository -y 'ppa:hvr/ghc'
             apt-get update
             apt-get install -y "$HCNAME"
             mkdir -p "$HOME/.ghcup/bin"
-            curl -sL https://downloads.haskell.org/ghcup/0.1.17.8/x86_64-linux-ghcup-0.1.17.8 > "$HOME/.ghcup/bin/ghcup"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.19.5/x86_64-linux-ghcup-0.1.19.5 > "$HOME/.ghcup/bin/ghcup"
             chmod a+x "$HOME/.ghcup/bin/ghcup"
-            "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
+            "$HOME/.ghcup/bin/ghcup" config add-release-channel https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-prereleases-0.0.7.yaml;
+            "$HOME/.ghcup/bin/ghcup" install cabal 3.10.1.0 || (cat "$HOME"/.ghcup/logs/*.* && false)
           fi
         env:
           HCKIND: ${{ matrix.compilerKind }}
@@ -135,24 +146,26 @@ jobs:
           echo "CABAL_CONFIG=$HOME/.cabal/config" >> "$GITHUB_ENV"
           HCDIR=/opt/$HCKIND/$HCVER
           if [ "${{ matrix.setup-method }}" = ghcup ]; then
-            HC=$HOME/.ghcup/bin/$HCKIND-$HCVER
+            HC=$("$HOME/.ghcup/bin/ghcup" whereis ghc "$HCVER")
+            HCPKG=$(echo "$HC" | sed 's#ghc$#ghc-pkg#')
+            HADDOCK=$(echo "$HC" | sed 's#ghc$#haddock#')
             echo "HC=$HC" >> "$GITHUB_ENV"
-            echo "HCPKG=$HOME/.ghcup/bin/$HCKIND-pkg-$HCVER" >> "$GITHUB_ENV"
-            echo "HADDOCK=$HOME/.ghcup/bin/haddock-$HCVER" >> "$GITHUB_ENV"
-            echo "CABAL=$HOME/.ghcup/bin/cabal-3.6.2.0 -vnormal+nowrap" >> "$GITHUB_ENV"
+            echo "HCPKG=$HCPKG" >> "$GITHUB_ENV"
+            echo "HADDOCK=$HADDOCK" >> "$GITHUB_ENV"
+            echo "CABAL=$HOME/.ghcup/bin/cabal-3.10.1.0 -vnormal+nowrap" >> "$GITHUB_ENV"
           else
             HC=$HCDIR/bin/$HCKIND
             echo "HC=$HC" >> "$GITHUB_ENV"
             echo "HCPKG=$HCDIR/bin/$HCKIND-pkg" >> "$GITHUB_ENV"
             echo "HADDOCK=$HCDIR/bin/haddock" >> "$GITHUB_ENV"
-            echo "CABAL=$HOME/.ghcup/bin/cabal-3.6.2.0 -vnormal+nowrap" >> "$GITHUB_ENV"
+            echo "CABAL=$HOME/.ghcup/bin/cabal-3.10.1.0 -vnormal+nowrap" >> "$GITHUB_ENV"
           fi
 
           HCNUMVER=$(${HC} --numeric-version|perl -ne '/^(\d+)\.(\d+)\.(\d+)(\.(\d+))?$/; print(10000 * $1 + 100 * $2 + ($3 == 0 ? $5 != 1 : $3))')
           echo "HCNUMVER=$HCNUMVER" >> "$GITHUB_ENV"
           echo "ARG_TESTS=--enable-tests" >> "$GITHUB_ENV"
           echo "ARG_BENCH=--enable-benchmarks" >> "$GITHUB_ENV"
-          if [ $((HCNUMVER >= 90400)) -ne 0 ] ; then echo "HEADHACKAGE=true" >> "$GITHUB_ENV" ; else echo "HEADHACKAGE=false" >> "$GITHUB_ENV" ; fi
+          if [ $((HCNUMVER >= 90800)) -ne 0 ] ; then echo "HEADHACKAGE=true" >> "$GITHUB_ENV" ; else echo "HEADHACKAGE=false" >> "$GITHUB_ENV" ; fi
           echo "ARG_COMPILER=--$HCKIND --with-compiler=$HC" >> "$GITHUB_ENV"
           echo "GHCJSARITH=0" >> "$GITHUB_ENV"
         env:
@@ -209,14 +222,14 @@ jobs:
       - name: install cabal-plan
         run: |
           mkdir -p $HOME/.cabal/bin
-          curl -sL https://github.com/haskell-hvr/cabal-plan/releases/download/v0.6.2.0/cabal-plan-0.6.2.0-x86_64-linux.xz > cabal-plan.xz
-          echo 'de73600b1836d3f55e32d80385acc055fd97f60eaa0ab68a755302685f5d81bc  cabal-plan.xz' | sha256sum -c -
+          curl -sL https://github.com/haskell-hvr/cabal-plan/releases/download/v0.7.3.0/cabal-plan-0.7.3.0-x86_64-linux.xz > cabal-plan.xz
+          echo 'f62ccb2971567a5f638f2005ad3173dba14693a45154c1508645c52289714cb2  cabal-plan.xz' | sha256sum -c -
           xz -d < cabal-plan.xz > $HOME/.cabal/bin/cabal-plan
           rm -f cabal-plan.xz
           chmod a+x $HOME/.cabal/bin/cabal-plan
           cabal-plan --version
       - name: checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           path: source
       - name: initial cabal.project for sdist
@@ -254,8 +267,8 @@ jobs:
         run: |
           $CABAL v2-build $ARG_COMPILER $ARG_TESTS $ARG_BENCH --dry-run all
           cabal-plan
-      - name: cache
-        uses: actions/cache@v2
+      - name: restore cache
+        uses: actions/cache/restore@v3
         with:
           key: ${{ runner.os }}-${{ matrix.compiler }}-${{ github.sha }}
           path: ~/.cabal/store
@@ -279,8 +292,14 @@ jobs:
           ${CABAL} -vnormal check
       - name: haddock
         run: |
-          $CABAL v2-haddock --haddock-all $ARG_COMPILER --with-haddock $HADDOCK $ARG_TESTS $ARG_BENCH all
+          $CABAL v2-haddock --disable-documentation --haddock-all $ARG_COMPILER --with-haddock $HADDOCK $ARG_TESTS $ARG_BENCH all
       - name: unconstrained build
         run: |
           rm -f cabal.project.local
           $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks all
+      - name: save cache
+        uses: actions/cache/save@v3
+        if: always()
+        with:
+          key: ${{ runner.os }}-${{ matrix.compiler }}-${{ github.sha }}
+          path: ~/.cabal/store

--- a/.github/workflows/macOS.yml
+++ b/.github/workflows/macOS.yml
@@ -29,11 +29,11 @@ jobs:
     runs-on: macos-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         submodules: recursive
 
-    - uses: haskell/actions/setup@v1
+    - uses: haskell-actions/setup@v2
       id: setup-haskell
       with:
         ghc-version:   latest
@@ -44,7 +44,7 @@ jobs:
         cabal update
         cabal configure -O1 --enable-tests
 
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3
       name: Cache dependencies
       id: cache
       with:
@@ -56,7 +56,7 @@ jobs:
     - name: Install dependencies
       if: ${{ !steps.cache.outputs.cache-hit }}
       run: |
-        cabal build --only-dependencies --enable-tests
+        cabal build --only-dependencies
 
     - name: Build
       run: |

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -29,11 +29,11 @@ jobs:
     runs-on: windows-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         submodules: recursive
 
-    - uses: haskell/actions/setup@v1
+    - uses: haskell-actions/setup@v2
       id: setup-haskell
       with:
         ghc-version:   latest
@@ -44,7 +44,7 @@ jobs:
         cabal update
         cabal configure -O1 --enable-tests
 
-    - uses: actions/cache@v2
+    - uses: actions/cache@v3
       name: Cache dependencies
       id: cache
       with:
@@ -56,7 +56,7 @@ jobs:
     - name: Install dependencies
       if: ${{ !steps.cache.outputs.cache-hit }}
       run: |
-        cabal build --only-dependencies --enable-tests
+        cabal build --only-dependencies
 
     - name: Build
       run: |

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,12 @@
 Changes
 =======
 
+Version 3.3.1.2 (10 Sep 2023)
+---------------
+
+* Fix compilation with `tasty-1.5`
+* Tested with GHC 7.4 - 9.8.0
+
 Version 3.3.1.1 (8 May 2022)
 ---------------
 

--- a/Test/Tasty/Silver/Interactive.hs
+++ b/Test/Tasty/Silver/Interactive.hs
@@ -578,7 +578,11 @@ produceOutput opts tree =
       foldTestTree
         trivialFold
           { foldSingle = handleSingleTest
-          , foldGroup = handleGroup
+#if MIN_VERSION_tasty(1,5,0)
+          , foldGroup = \ opts name -> foldMap (handleGroup opts name)
+#else
+          , foldGroup = \ opts name -> id      (handleGroup opts name)
+#endif
           }
           opts tree
 
@@ -917,22 +921,34 @@ instance Ord a => Monoid (Maximum a) where
   mempty = MinusInfinity
   mappend = (<>)
 
-
--- | Compute the amount of space needed to align "OK"s and "FAIL"s
+-- | Compute the amount of space needed to align "OK"s and "FAIL"s.
+--
 computeAlignment :: OptionSet -> TestTree -> Int
 computeAlignment opts =
-  fromMonoid .
-  foldTestTree
-    trivialFold
-      { foldSingle = \_ name _ level -> Maximum (length name + level)
-      , foldGroup = \_ _ m -> m . (+ indentSize)
-      }
-    opts
+  fromMonoid . foldTestTree f opts
   where
+    fromMonoid :: (Int -> Maximum Int) -> Int
     fromMonoid m =
       case m 0 of
         MinusInfinity -> 0
         Maximum x -> x
+
+    f :: TreeFold (Int -> Maximum Int)
+    f = trivialFold
+      { foldSingle = \ _opts  name _test level -> addName   name level
+      , foldGroup  = \ _opts _name group level -> addIndent level group
+      }
+
+    addName :: TestName -> Int -> Maximum Int
+    addName name level = Maximum $ length name + level
+
+#if MIN_VERSION_tasty(1,5,0)
+    addIndent :: Int -> [Int -> Maximum Int] -> Maximum Int
+    addIndent level = foldMap ($ (level + indentSize))
+#else
+    addIndent :: Int -> (Int -> Maximum Int) -> Maximum Int
+    addIndent level = id      ($ (level + indentSize))
+#endif
 
 -- (Potentially) colorful output
 ok, warn, failure, infoOk, infoWarn, infoFail :: (?colors :: Bool) => String -> IO ()

--- a/tasty-silver.cabal
+++ b/tasty-silver.cabal
@@ -29,8 +29,10 @@ extra-source-files:
   README.md
 
 tested-with:
-  GHC == 9.4.1
-  GHC == 9.2.3
+  GHC == 9.8.0
+  GHC == 9.6.2
+  GHC == 9.4.7
+  GHC == 9.2.8
   GHC == 9.0.2
   GHC == 8.10.7
   GHC == 8.8.4

--- a/tasty-silver.cabal
+++ b/tasty-silver.cabal
@@ -1,6 +1,6 @@
 cabal-version:       1.14
 name:                tasty-silver
-version:             3.3.1.1
+version:             3.3.1.2
 synopsis:            A fancy test runner, including support for golden tests.
 description:
   This package provides a fancy test runner and support for «golden testing».
@@ -48,7 +48,7 @@ tested-with:
 
 Source-repository head
   type:     git
-  location: git://github.com/phile314/tasty-silver.git
+  location: https://github.com/phile314/tasty-silver.git
 
 library
   Default-language:

